Add Ascend NPU entry (#2080)
pytorch/pytorch.github.io: _get_started/additional_platforms/ascend_npu.md

# Installing on Ascend NPU

> Ascend NPU is Huawei's AI processor series. The Ascend Extension for PyTorch (`torch_npu`) enables PyTorch to run on Ascend NPUs, supporting eager-mode execution, distributed training, and mixed precision via the `PrivateUse1` backend.

Reviewer: ```suggestion
**Ascend for PyTorch (`torch-npu`)** is a deep learning adaptation framework built on Ascend. It enables Ascend NPUs to support the PyTorch framework, delivering the powerful computing capabilities of Ascend AI processors to PyTorch developers and users.
```

> **Ascend for PyTorch (`TorchNPU`)** is a deep learning adaptation framework built on Ascend. It enables Ascend NPUs to support the PyTorch framework, delivering the powerful computing capabilities of Ascend AI processors to PyTorch developers and users.

Reviewer: ```suggestion
**Ascend for PyTorch (TorchNPU)** is a deep learning adaptation framework built on Ascend. It enables Ascend NPUs to support the PyTorch framework, delivering the powerful computing capabilities of Ascend AI processors to PyTorch developers and users.
```

We use `` for code in most cases, but TorchNPU is an alias, so we don't need `` to embrace it.

## Prerequisites

### Hardware Requirements

> * Huawei Ascend NPU(s), for example, 910B, 910C, or 310P processor

Reviewer: ```suggestion
* Ascend for PyTorch supports most Ascend platforms. Please check the [hardware compatibility](https://www.hiascend.com/hardware/compatibility) before installing.
```

> * Ascend for PyTorch supports most Ascend platforms. Please check the [hardware compatibility](https://www.hiascend.com/hardware/compatibility) before installing.

Reviewer 2: https://www.hiascend.com/en/hardware/compatibility/
This page is exists, but no content in it.

### Software Requirements

* Python >= 3.10
* CANN (Compute Architecture for Neural Networks) toolkit installed >= 8.0
* Ascend driver and firmware installed

> Before installing PyTorch with Ascend NPU support, you must install the CANN toolkit. Download it from the [Ascend Community](https://www.hiascend.com/) and follow the [CANN Installation Guide](https://www.hiascend.com/cann/download?versionId=735&ids=d806%2Ch0501%2Ch0601%2Ch0702).

Reviewer 2: Use English version of these pages.
https://www.hiascend.com/en
hiascend.com/en/cann/download

## Installation

### pip

```bash
pip3 install torch==2.10.0 --index-url https://download.pytorch.org/whl/cpu && pip3 install torch-npu==2.10.0
```

> Use the pip package manager to install PyTorch with Ascend NPU support. Please refer to the [installation page](https://www.hiascend.com/developer/software/ai-frameworks/pytorch/download?versionId=167&ids=26958bcc909e4cd48fa56d4c4a43ebec%2C98%2C106%2C1%2C6%2C3%2C) and select your preferred options in the selector above to get the installation command.

Reviewer 2: https://www.hiascend.com/en/developer/software/ai-frameworks/pytorch/download

```python
import torch
import torch_npu

print(torch.__version__)
print("torch_npu version:", torch_npu.__version__)

if torch.npu.is_available():
    print("Ascend NPU is available!")
    print(f"NPU count: {torch_npu.npu.device_count()}")
    print(f"NPU name: {torch_npu.npu.get_device_name(0)}")

    # Test basic computation
    a = torch.randn(3, 4).npu()
    b = torch.randn(3, 4).npu()
    print("NPU computation result:", a + b)
else:
    print("Ascend NPU is not available.")
```

> Expected output should show the PyTorch and torch_npu versions, NPU count, device name, and the result of a tensor addition without errors.

Reviewer: ````suggestion
```python
import torch
import torch_npu

x = torch.randn(2, 2).npu()
y = torch.randn(2, 2).npu()
z = x.mm(y)

print(z)
```

The following similar output indicates successful installation:

```bash
tensor([[-0.0515,  0.3664],
        [-0.1258, -0.5425]],  device='npu:0')
```
````

## Documentation

For more information, please visit:

> * [Ascend for PyTorch Official Documentation](https://www.hiascend.com/document/detail/zh/Pytorch/2600/index/index.html)

Reviewer 2: EN: https://www.hiascend.com/document/detail/en/Pytorch/latest/index/index.html
ZH: https://www.hiascend.com/document/detail/zh/Pytorch/latest/index/index.html

> * [Ascend for PyTorch Official Documentation](https://www.hiascend.com/document/detail/zh/Pytorch/2600/index/index.html)

Reviewer 2: It's better to add PyTorch Portal
EN: https://www.hiascend.com/en/developer/software/ai-frameworks/pytorch
ZH: https://www.hiascend.com/cn/developer/software/ai-frameworks/pytorch

> * [Ascend Community Portal](https://www.hiascend.com/)

Reviewer 2: I think we should provide CANN main page, for imformation and install.
EN: https://www.hiascend.com/eng/cann
ZH: https://www.hiascend.com/cann

> * [Ascend Community Portal](https://www.hiascend.com/)

Reviewer 2: Ascend Community Portal: 
EN: https://www.hiascend.com/en
ZH: https://www.hiascend.com/

## Prerequisites

### Hardware Requirements

* Ascend for PyTorch supports most Ascend platforms. Please check the hardware compatibility before installing.

### Software Requirements

* Python >= 3.10
* CANN (Compute Architecture for Neural Networks) toolkit installed >= 8.0
* Ascend driver and firmware installed

Before installing PyTorch with Ascend NPU support, you must install the CANN toolkit. Download it from the [Ascend Community](https://www.hiascend.com/en) and follow the [CANN Installation Guide](https://www.hiascend.com/en/cann/download).

## Installation

### pip

```bash
pip3 install torch==2.10.0 --index-url https://download.pytorch.org/whl/cpu && pip3 install torch-npu==2.10.0
```

Use the pip package manager to install PyTorch with Ascend NPU support. Please refer to the [installation page](https://www.hiascend.com/en/developer/software/ai-frameworks/pytorch/download) and select your preferred options in the selector above to get the installation command.

## Verification

To ensure that PyTorch was installed correctly with Ascend NPU support, run the following code:

```python
import torch
import torch_npu

x = torch.randn(2, 2).npu()

> y = torch.randn(2, 2).npu()

Reviewer 3: nit: we usually recommend users do `torch.randn(2, 2, device="npu")` to avoid the extra cpu->accelerator copy

z = x.mm(y)

print(z)
```

> The following similar output indicates successful installation:

Reviewer 3: ```suggestion
The following, or a similar output, indicates successful installation:
```

# Installing on Ascend NPU

Ascend NPU is Huawei's AI processor series (including Ascend 910B, 910C, and 310P). The Ascend Extension for PyTorch (`torch_npu`) enables PyTorch to run on Ascend NPUs, supporting eager-mode execution, distributed training, and mixed precision via the `PrivateUse1` backend.

## Prerequisites

### Hardware Requirements

* Ascend 910B, 910C, or 310P processor

### Software Requirements

* Python >= 3.10
* CANN (Compute Architecture for Neural Networks) toolkit installed >= 8.0
* Ascend driver and firmware installed

Before installing PyTorch with Ascend NPU support, you must install the CANN toolkit. Download it from the [Ascend Community](https://www.hiascend.com/) and follow the [CANN Installation Guide](https://www.hiascend.com/cann/download?versionId=735&ids=d806%2Ch0501%2Ch0601%2Ch0702).

## Installation

### pip

Use the pip package manager to install PyTorch with Ascend NPU support. Select your preferred options in the selector above to get the installation command.

After installation, source the CANN environment:

```bash
source <ASCEND_INSTALL_DIR>/Ascend/cann/set_env.sh
source <ASCEND_INSTALL_DIR>/Ascend/nnal/atb/set_env.sh

> ```

Reviewer: This is CANN installation steps, not for torch-npu. Should be `pip install torch-npu` here. Please refer to https://www.hiascend.com/developer/software/ai-frameworks/pytorch/download?versionId=167&ids=26958bcc909e4cd48fa56d4c4a43ebec%2C98%2C106%2C1%2C6%2C3%2C

Author: Ok. Fixed.
